File: Data Structures & Algorithms/implement-stack-using-queues/submission-1.java
class MyStack {
    private ArrayDeque<Integer> q1;
    private ArrayDeque<Integer> q2;

    public MyStack() {
        q1 = new ArrayDeque<>();
        q2 = new ArrayDeque<>();
    }
    
    public void push(int x) {
        q1.add(x);
    }
    
    public int pop() {
        int temp;
        while (q1.size() > 1) {
            temp = q1.remove();
            q2.add(temp);
        }
        temp = q1.remove();
        ArrayDeque<Integer> tempQueue = q1;
        q1 = q2;
        q2 = tempQueue;
        return temp;
    }
    
    public int top() {
        return q1.peekLast();
    }
    
    public boolean empty() {
        if (q1.size() <= 0)
            return true;
        else 
            return false;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: Data Structures & Algorithms/time-based-key-value-store/submission-1.py
class TimeMap:
    def __init__(self):
        self.time_map = {}

    def set(self, key: str, value: str, timestamp: int) -> None:
        if key not in self.time_map:
            self.time_map[key] = []

        self.time_map[key].append((value, timestamp))
        # print(self.time_map)

    def get(self, key: str, timestamp: int) -> str:
        if key not in self.time_map:
            return ""

        # print(self.time_map)
        # print(self.time_map[key])
        value_timestamp_list = self.time_map[key]
        low = 0
        high = len(value_timestamp_list) - 1
        index = 0
        found = False

        while low <= high:
            mid = (low + high) // 2
            if value_timestamp_list[mid][1] <= timestamp:
                found = True
                index = mid

                if mid != high and value_timestamp_list[mid + 1][1] > timestamp:
                    break

                low = mid + 1
            else:
                high = mid - 1

        if not found:
            return ""

        return value_timestamp_list[index][0]




# Your TimeMap object will be instantiated and called as such:
# obj = TimeMap()
# obj.set(key,value,timestamp)
# param_2 = obj.get(key,timestamp)